### ISSUE TITLE
"system_tests/run_command" unit test fails on MSVC

### ISSUE THREAD

**Issue by vostrnad** [bitcoin/bitcoin#26368]
I'm building v24.0rc2 on Windows 10 using Visual Studio 2022. The build succeeds without errors and the binary works fine, but there is one failing unit test:

```
C:\bitcoin\src>test_bitcoin.exe
Running 528 test cases...
C:/bitcoin/src/test/system_tests.cpp(64): error: in "system_tests/run_command": check what.find(expected) != std::string::npos has failed

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
```

Also tried with v23.0 and latest master, same error (for v23.0 the line number is different).

**Comment by hebasto** [comment]
@vostrnad 

What is Boost version you are using to compile your binaries?

**Comment by vostrnad** [comment]
@hebasto Thank you for your response. I install dependencies with vcpkg as per the MSVC build instructions, so I assume my Boost version will be correct, but I don't know how to check what version I'm using.

The release unit test binary has no failing tests, but it says `Running 527 test cases...` instead of 528. When I compare the outputs of running `test_bitcoin.exe -l test_suite`, the `run_command` test case is not present in the release binary output:
```
test/system_tests.cpp(24): Entering test suite "system_tests"
test/system_tests.cpp(28): Entering test case "dummy"
test/system_tests.cpp(28): Leaving test case "dummy"; testing time: 12072us
test/system_tests.cpp(24): Leaving test suite "system_tests"; testing time: 15387us
```
vs
```
C:\bitcoin\src\test\system_tests.cpp(24): Entering test suite "system_tests"
C:\bitcoin\src\test\system_tests.cpp(28): Entering test case "dummy"
C:\bitcoin\src\test\system_tests.cpp(28): Leaving test case "dummy"; testing time: 9513us
C:\bitcoin\src\test\system_tests.cpp(35): Entering test case "run_command"
C:/bitcoin/src/test/system_tests.cpp(64): error: in "system_tests/run_command": check what.find(expected) != std::string::npos has failed
C:\bitcoin\src\test\system_tests.cpp(35): Leaving test case "run_command"; testing time: 58694us
C:\bitcoin\src\test\system_tests.cpp(24): Leaving test suite "system_tests"; testing time: 80393us
```

My Windows is for the most part not localized. I have English (US) and Czech set as preferred languages (in this order, so almost everything is in English) and use the Czech regional format.

**Comment by hebasto** [comment]
@vostrnad 

Could you please make another build with a diff as follows:
```diff
--- a/src/test/system_tests.cpp
+++ b/src/test/system_tests.cpp
@@ -59,7 +59,7 @@ BOOST_AUTO_TEST_CASE(run_command)
         BOOST_CHECK_EXCEPTION(RunCommandParseJSON("invalid_command"), boost::process::process_error, [&](const boost::process::process_error& e) {
             const std::string what(e.what());
             BOOST_CHECK(what.find("RunCommandParseJSON error:") == std::string::npos);
-            BOOST_CHECK(what.find(expected) != std::string::npos);
+            BOOST_CHECK_EQUAL(what, std::string());
             return true;
         });
     }
```

then run `test_bitcoin.exe --run_test=system_tests/run_command`, and post the resulted log with an error?

**Comment by vostrnad** [comment]
@hebasto 
```
C:\bitcoin\src>test_bitcoin.exe --run_test=system_tests/run_command
Running 1 test case...
C:/bitcoin/src/test/system_tests.cpp(64): error: in "system_tests/run_command": check what == std::string() has failed [ CreateProcess failed: Systm neme nalzt uveden soubor. != ]

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
```
So it seems some Czech localization does leak out, notably with all characters with diacritics removed. The actual sentence would be "Systém nemůže nalézt uvedený soubor" which means "The system cannot find the file specified".

**Comment by hebasto** [comment]
So MSVC-compiled `test_bitcoin.exe` requires non-localized environment.

Perhaps, it is worth mentioning somewhere in our docs.

Closing.

**Comment by hebasto** [comment]
@vostrnad 

Mind testing a fix suggested in ##26377?

**Comment by vostrnad** [comment]
@hebasto Not at all. Can confirm this fixes the problem for me.
